frontend/freebase2wikidata.js

```diff
@@ -260,8 +260,85 @@ $(function() {
     var title = mw.config.get('wgTitle');
     return qidRegEx.test(title) ? title : false;
   }
-  
-  (function generateNav() { 
+
+    ps.openNav = function openNav(itemLabel, propertyLabel, propertyValue, referenceURL, buttons) {
+        $('#myNav').width('100%');
+
+        var blackboard = $('#blackboard');
+
+        blackboard.append(buttons.eq(0).clone(true, true));
+        blackboard.append(buttons.eq(1).clone(true, true));
+
+        blackboard.append('<div class="loader"></div>');
+
+        $.ajax({
+            type: 'GET',
+            url: 'https://tools.wmflabs.org/strephit/search?url=' + encodeURI(referenceURL),
+            success: function(msg) {
+                $('.loader').remove();
+
+                if(msg != 'no preview'){
+                    var data = JSON.parse(msg);
+                    var regEx = new RegExp('(' + itemLabel + '|' + propertyLabel + '|' + propertyValue + ')(?!([^<]+)?>)', 'gi');
+
+                    if(!data.hasOwnProperty('excerpt'))
+                    {
+                        blackboard.append('<h1><a href="' + data.url + '">' + data.url + '</a></h1>');
+                        blackboard.append('<ul></ul>');
+                        
+                        var linklist = $('#blackboard > ul');
+                        for(var index in data){
+                            var item = data[index];
+                            if(index != 'url' && item !== null){
+                                if( index == 'other' ) {
+                                    while(typeof item == 'string'){
```

Comment:
Seems quite dangerous (but it's maybe what should be done)

Reply:
I can see your point, but atm I'm not able to provide a better solution for this, as I don't have a consistent format for the data I get from the api.